src/saltext/vmware/utils/drift.py

```diff
@@ -0,0 +1,100 @@
+# Copyright 2021 VMware, Inc.
+# SPDX-License-Identifier: Apache-2.0
+"""
+Common functions used across modules
+"""
+import json
+
+
+def drift_report(obj1, obj2, diff_level=None):
+    """
+    Finds the drift between two objects/dictionaries
+
+    obj1
+        First dictionary
+    obj2
+        Second dictionary
+    diff_level
+        Defines on what tree level to show the drift.
+        If not specified the changes are shown in the last leaf level.
+    """
+    result_diffs = []
+    _drift_recurse_(obj1, obj2, result_diffs)
+
+    result_tree = {}
+    for tup in result_diffs:
+        _drift_tree_(tup, result_tree)
+
+    if diff_level is not None and isinstance(result_tree, dict):
```

Comment:
**suggestion (non-blocking)** Not sure if `diff_level` can be 0 (would be useful to mention in the docstring), but if it can only be None or a non-zero amount, then `if diff_level and isinstance(result_tree, dict)` would do the job.

Otherwise, yeah, this looks good.

Reply:
I added more info about the logic in the doc of the method

Reply:
done